### PR TITLE
Add svg-as-string mode

### PR DESCRIPTION
**Pull request by vhf**
Here is a suggestion, I'd love some feedback.

Motivation: in some situations I'd rather output the SVG directly instead of serving it as an asset.

### CODE REVIEW

**Comment by brendo** [comment]
Hey! 👋 

This is cool idea, I like it and sometimes we've wanted to do the same.

How do you feel about exposing the behaviour as a plugin option, rather than a vFile property?

Travis isn't happy with the build, I'll take a look at the underlying `mermaid.cli` and see if there's something that can be changed. Puppeteer recently went v1, so it may resolve the issue.

**Comment by vhf** [comment]
Agreed, it'd be better as a plugin option.

**Comment by vhf** [comment]
Done. 😄

File: jest.config.js [hunk 1]
```diff
@@ -5,7 +5,6 @@ module.exports = {
   coveragePathIgnorePatterns: [
     '/node_modules/',
   ],
-  mapCoverage: true,
```
Comment (vhf) [inline jest.config.js:8]: https://github.com/facebook/jest/issues/4509#issuecomment-367410344

**Comment by vhf** [comment]
@brendo Tests are still passing locally and still failing on Travis, I tried pushing updates for mermaid.cli (latest uses puppeteer 1.x), it didn't help. Any idea? :)

**Comment by artragis** [comment]
Hello @brendo do you have some news about that?

**Comment by jimthedev** [comment]
Hey all, any update to this? If not would it be ok if someone picked it up? Wouldn't want to step on toes.

**Comment by vhf** [comment]
@jimthedev Feel free to take over my initial attempt!

**Comment by hugmanrique** [comment]
Any updates on this? I don't fully understand how puppeter works, so I'm not really able to pick this up and fix the test.

File: src/index.js [hunk 2]
```diff
@@ -90,7 +90,7 @@ function replaceLinkWithEmbedded(node, index, parent, vFile) {
  * @param {boolean} isSimple
```
Comment (brendo) [inline src/index.js:90]: Minor, JSDoc update now that this an object that has two possible properties (there's a few other updates needed around this as well).

File: test/index.test.js [hunk 3]
```diff
@@ -40,6 +41,17 @@ describe('remark-mermaid', () => {
     expect(vfile.messages[0].message).toBe('mermaid code block replaced with graph');
   });
 
+  it('can handle code blocks to svg string', () => {
```
Comment (brendo) [inline test/index.test.js:44]: I'm curious if the failing test is timing related. If this is pulled up before the previous "can handle code blocks" test, does this one pass and the old test fail?

File: src/utils.js [hunk 4]
```diff
@@ -30,15 +30,21 @@ function render(source, destination) {
   // Clean up temporary file
   fs.removeSync(mmdPath);
 
+  if (asString) {
+    const string = fs.readFileSync(svgPath, { encoding: 'utf-8' });
+    fs.removeSync(svgPath);
```
Comment (brendo) [inline src/utils.js:35]: I wonder if this is the cause of the test errors 🤔

**Review by brendo — COMMENTED** [review]
:wave: I no longer work at Temando so can't help in terms of merging (@quangkhoa might be able to though?), but I've left a few comments.

**Comment by hugmanrique** [comment]
Thank you for still helping on a project you don't own anymore ❤

**Comment by yuku** [comment]
~~@brendo~~ @quangkhoa Cloud you trigger CI again? I just forked this branch for debugging, and finally, I found that all tests pass now!

https://travis-ci.org/yuku/remark-mermaid/builds/493032690

EDIT: I simply pushed after checking out this branch. I changed nothing.

**Comment by brendo** [comment]
Sorry @yuku, I no longer work at Temando so don't have access to do so.

@zomble, @nfour or @zyrorl?

**Comment by remcohaszing** [comment]
Why was this PR closed?

From the comments I understand this PR would work if CI was triggered.

I’m still interested in this feature. If this PR needs more changes, please give some feedback.

**Comment by vhf** [comment]
@remcohaszing It's been 16 months since I opened this and I gave up, feel free to take over: https://github.com/temando/remark-mermaid/pull/5#issuecomment-442519861 if you wish to push this forward, at this point I don't. :)

**Pull request closed without merging**